### PR TITLE
Js 14 create station modal window

### PR DESCRIPTION
**Pull request by Alekseiv25**
[Task link](https://jslab.atlassian.net/browse/JS-14?atlOrigin=eyJpIjoiZDQ2N2E3ZDBiNWY0NDI0ZWEyZDU3MmViNmFlZDM4ZjUiLCJwIjoiaiJ9)
Screenshot 2023-09-13 at 13 25 22
Screenshot 2023-09-13 at 13 25 34
Screenshot 2023-09-13 at 13 26 09

Screenshot 2023-09-14 at 16 48 19
Screenshot 2023-09-14 at 16 47 00
Screenshot 2023-09-14 at 16 47 16
Screenshot 2023-09-14 at 16 47 31
Screenshot 2023-09-14 at 16 47 57
Screenshot 2023-09-14 at 16 48 10

Create modal window with list of steps.
Add first step: Select a Station Type.
Add Adaptive version.

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -30,6 +30,11 @@
     "@typescript-eslint/no-shadow": "off",
     "react/require-default-props": "off",
     "no-confusing-arrow": "off",
-    "react/jsx-props-no-spreading": "off"
+    "react/jsx-props-no-spreading": "off",
+    "operator-linebreak": "off",
+    "react/no-array-index-key": "off",
+    "no-console": "off",
+    "no-param-reassign": "off",
+    "import/no-extraneous-dependencies": "off"
```
Comment (alexandragul) [inline .eslintrc.json:38]: Why did you turn off all this rules? There are a lot of helpful ones

File: src/components/common/Button/index.tsx [hunk 2]
```diff
@@ -26,6 +28,8 @@ const StyledInput = styled.input<StyledProps | IButtonProps | SpaceProps>`
   gap: 10px;
   cursor: pointer;
   ${space}
+  border: none;
+  outline: ${({ isSelected }) => (isSelected ? '3px solid black' : 'none')};
```
Comment (alexandragul) [inline src/components/common/Button/index.tsx:32]: Use color from the system

File: src/components/common/Button/index.tsx [hunk 3]
```diff
@@ -37,13 +41,14 @@ const StyledInput = styled.input<StyledProps | IButtonProps | SpaceProps>`
   ${variant({
     variants: {
       primary: {
-        border: 0,
         borderRadius: '22px',
+        padding: '10px 24px',
```
Comment (alexandragul) [inline src/components/common/Button/index.tsx:45]: Why didn't you use spacing from the system here?

File: src/components/common/Button/index.tsx [hunk 4]
```diff
@@ -37,13 +41,14 @@ const StyledInput = styled.input<StyledProps | IButtonProps | SpaceProps>`
...
       },
       secondary: {
+        padding: '16px 126px',
         borderRadius: '16px',
+        fontWeight: `${theme.fontWeights.medium}`,
```
Comment (alexandragul) [inline src/components/common/Button/index.tsx:50]: Does it works?

```suggestion
        fontWeight: `${props => props.theme.fontWeights.medium}`,
```

File: src/components/modal/ModalStation/SelectStation/index.tsx [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+`;
+
+const StyledForm = styled(Form)`
+  display: flex;
+  flex-direction: column;
+  height: 100%;
+`;
+
+function SelectStationContainer(props: ISelectStationProps) {
+  const { data, next } = props;
```
Comment (alexandragul) [inline src/components/modal/ModalStation/SelectStation/index.tsx:33]: It's a better practise to follow this pattern

```suggestion
const SelectStationContainer: React.FC<ISelectStationProps> => ({ data, next }) {
```

**Review by Andreushk — APPROVED** [review]

File: src/components/modal/ModalStation/SelectStation/index.tsx [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+      <Formik initialValues={data} onSubmit={handleSubmit}>
+        {() => (
+          <StyledForm>
+            <StyledButtonsContainer mb={5}>
+              <FieldButton
+                name="type"
+                value="Gasoline / Diesel"
+                type="button"
+                variant="secondary"
+                isSelected={selectedButton === 'Gasoline / Diesel'}
```
Comment (alexandragul) [inline src/components/modal/ModalStation/SelectStation/index.tsx:61]: Please create a constant for this selections. Ex.

```
enum StationType {
  GASOLINE = 'Gasoline',
  ELECTRIC = 'Electric',
  ...
}
```

File: src/components/modal/ModalStation/index.tsx [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+    'Connect your Bank',
+    'Connect your System',
+    'Finalize',
+  ];
+
+  const handleNextStep = (newData: Partial<IStationData>) => {
+    setData((prev) => ({ ...prev, ...newData }));
+    setCurrentStep((prev) => prev + 1);
+  };
+  console.log(data);
```
Comment (alexandragul) [inline src/components/modal/ModalStation/index.tsx:27]: Please never leave logs in code

File: src/components/modal/ModalStation/index.tsx [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+  const [currentStep, setCurrentStep] = useState(0);
+
+  const navItems = [
+    'Station Type',
+    'General Information',
+    'Station Amenities',
+    'Connect your Bank',
+    'Connect your System',
+    'Finalize',
+  ];
```
Comment (alexandragul) [inline src/components/modal/ModalStation/index.tsx:21]: Object is more readable and easy to handle
```suggestion
  enum NavItems = {
    STATION_TYPE = 'Station Type',
    ...
  }
```

File: src/constants/stationData.ts [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+  tabacco: false,
+  lottery: false,
+  carWash: false,
+  restrooms: false,
+  ATM: false,
+  foodOfferings: false,
+  restaurant: false,
+  overnightParking: false,
+  showers: false,
+};
```
Comment (alexandragul) [inline src/constants/stationData.ts:25]: Do you need it all over the app? Why do you create the constant on a top level?

**Review by alexandragul — COMMENTED** [review]
Please when you finish this PR, create new one for mobile version

File: src/components/modal/ModalStation/index.tsx [hunk 10]
```diff
@@ -0,0 +1,78 @@
...
+
+  const steps = [<SelectStation next={handleNextStep} data={data} />];
+
+  return (
+    <ModalWindow>
+      <ModalWindowNavigationContainer>
+        <Title variant="headerS" as="h3">
+          Add a Station
+        </Title>
+        {renderNavigationItems()}
```
Comment (alexandragul) [inline src/components/modal/ModalStation/index.tsx:71]:
```suggestion
    {Object.values(NavItems).map((value, index) => (
      <ModalWindowNavigationListItem
        key={uuidv4()}
        index={index + 1}
        value={value}
        variant={currentStep === index ? 'selected' : 'notSelected'}
      />
    ))}
```

File: src/store/users/users.api.ts [hunk 11]
```diff
@@ -1,10 +1,12 @@
...
+const usersApi = createApi({
   reducerPath: 'users',
   baseQuery: fetchBaseQuery({
     baseUrl: 'https://localhost:5000',
   }),
   tagTypes: ['User'],
   endpoints: () => ({}),
 });
+
+export default usersApi;
```
Comment (alexandragul) [inline src/store/users/users.api.ts:12]: I don't see a real benefit of default exports for utils. It's usually implemented just for components

**Review by alexandragul — APPROVED** [review]

File: src/store/users/users.api.ts (hunk 11 above)
Comment (Alekseiv25) [reply src/store/users/users.api.ts:12]: These are eslint requirements that the component be exported if it is single

**Pull request merged**